### PR TITLE
[HttpKernel][DataCollectorInterface] Ease compatibility

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/pull/33065/files#r342109866
| License       | MIT
| Doc PR        | -

To ease data collectors being compatible from 4.3 to 5.0.

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/DataCollector/DataCollectorInterface.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ interface DataCollectorInterface
     /**
      * Collects data for the given Request and Response.
      */
-    public function collect(Request $request, Response $response, \Exception $exception = null);
+    public function collect(Request $request, Response $response/*, \Exception $exception = null*/);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DataCollectorInterface.php]: this must be updated to `Throwable` and a corresponding annotation should be added

**Review by nicolas-grekas — APPROVED** [review]
~(once comment is addressed)~

This is [allowed by the BC promise](https://symfony.com/doc/current/contributing/code/bc.html#id67) - and is the only way to make this argument accept `Throwable` in Symfony 5.0, while still allowing third-party bundles to be compatible with 3.4 (since most are as of now)

**Comment by nicolas-grekas** [comment]
After talking with @fabpot, I'm going to merge this change on 4.3.

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Comment by teohhanhui** [comment]
> This is allowed by the [BC promise](https://symfony.com/doc/current/contributing/code/bc.html#id67)

Actually, the note on the BC promise page is inaccurate. Only an optional argument could be removed:

Works: https://3v4l.org/voGVW
Does not work: https://3v4l.org/C7SD4

**Comment by nicolas-grekas** [comment]
Correct, can you please sent a PR to the corresponding doc page?

**Comment by alcaeus** [comment]
The fact that this change wasn't merged into 3.4 makes supporting Symfony 3.4 in a bundle unnecessarily complicated: Symfony 5 and Symfony 3.4 can't be supported in the same version due to this incompatibility. @fabpot what was the reason to only merge this into 4.3, but not 3.4?

**Comment by fabpot** [comment]
I should have merged it in 4.4 as this is not a bug fix. I agreed to merge it in 4.3 to ease the migration path. Our goal is that third-parties can support 2 consecutive versions (3 and 4 OR 4 and 5). Supporting 3, 4, and 5 is not possible.

**Pull request merged**